### PR TITLE
[jackett] fix #210 speed up search

### LINKED ISSUE #210: Jackett can't do multiple searches at the same time
the jackett repo redirected me here 
https://github.com/Jackett/Jackett/issues/13551

### PR DESCRIPTION
**Pull request by galeksandrp**
[jackett] fix #210 speed up search

### Issue

`Jackett` from version [`v0.10.589` Jan 2, 2019](https://github.com/Jackett/Jackett/commit/25ddcb10bbf6e395a6cb33a8d10785c085e363ba) recommends separate requests to indexers instead of single request to `all` indexer.

Reason to that is because `Jackett` uses `Torznab` specification, which returns results serialized in single `XML`.
This means `all` indexer must wait for all indexers to finish before providing ANY results. This limits `all` indexer to the speed of slowest indexer.

### Solution

Finally rework `search` function to:
1. Retrieve list of configured indexers.
2. Make requests to indexers in parallel threads.
`Multithreading` is chosen instead of `multiprocessing` because we already in separate process thanks to `nova2.py` `multiprocessing` `main()`. Doing `multiprocessing` in `multiprocessing` is questionable matter and actually not working for me (`python 3.11.0` `Windows 11`).

### How to test
Install
```
https://raw.githubusercontent.com/galeksandrp/search-plugins/jackett-multi/nova3/engines/jackett.py
```
on top of `jackett.py 3.5` official plugin. Settings will be preserved.

### Tested
<details>
<summary>`jackett 0.20.2264 click to details`</summary>

image
</details>

machine | `python` | `OS` | `qBittorrent` | status
--- | --- | --- | --- | ---
`1` `ASUS` | `python 3.11.3` | `Windows 11 Home` | `qBittorrent 4.5.2` | `CLI` :ok: `GUI` 🆗 
`2` `HP` | `python 3.11.3` | `Arch Linux` | `qBittorrent 4.5.2` | `GUI` 🆗 `VueTorrent 1.5.6` 🆗 
`3` `PC` | `python 3.11.3` | `Windows 11 Pro` | `qBittorrent 4.5.2` | `GUI` 🆗 `VueTorrent 1.5.6` 🆗

### CODE REVIEW

**Comment by WojtekKowaluk** [comment]
Just implemented similar improvement, came here to make PR and found yours. It is almost perfect, but one thing should be fixed.

File: nova3/engines/jackett.py [hunk 1]
```diff
@@ -87,23 +93,59 @@ def search(self, what, cat='all'):
...
+
+        arguments = []
+        for indexer in xml.etree.ElementTree.fromstring(response).findall('indexer'):
+            arguments.append({'what': what,
+                              'jackettIndexerName': indexer.attrib['id'],
+                              'cat': cat})
+
+        if THREADED:
+            with Pool(min(len(arguments), MAX_THREADS)) as pool:
+                pool.map(self.jackettSearch, arguments)
```
[the remark this answers is not in the document]
Comment (galeksandrp) [reply nova3/engines/jackett.py:115]: @WojtekKowaluk Implemented "temporary" solution that you suggested.

[the remark this answers is not in the document]
Comment (WojtekKowaluk) [reply nova3/engines/jackett.py:115]: Nice, what was fast :)

**Comment by ngosang** [comment]
Closed by #239
Thank you both!

**Pull request closed without merging**